lib/block.rb

```diff
@@ -0,0 +1,37 @@
+require 'date'
+require 'pry'
+
+class Block
+  attr_reader :id, :reservations
+  
+  def initialize(id:, reservations: [])
+    @id = id
+    @reservations = reservations
+  end
+  
+  # Wave 3: check if a block has rooms available
+  def rooms_available
+    avail_block_reservations = []
+    
+    reservations.each do |reservation|
+      if reservation.block_res_taken == false
+        avail_block_reservations << reservation
+      end
+    end
+    
+    if avail_block_reservations.length == 0
+      raise ArgumentError.new("All rooms in this block are reserved.")
```

Comment:
This isn't a problem with the arguments, so don't raise an argument error. (in fact, there are no arguments)

Raise a different error.

Reply:
Ahhhhh. I've got this now!